Coin: add reading from and writing to `ByteBuffer`

schildbach opened the pull request:
Includes a test.

msgilligan commented on core/src/main/java/org/bitcoinj/core/FeeFilterMessage.java:
```diff
@@ -47,7 +47,7 @@ protected void bitcoinSerializeToStream(OutputStream stream) throws IOException
     @Override
     protected void parse() throws ProtocolException {
-        feeRate = Coin.ofSat(readUint64().longValue());
+        feeRate = Coin.read(payload);
```
I would go with: `Coin.ofSat(readInt64())` and not add the `ByteBuffer` stuff to `Coin`.

msgilligan reviewed (changes requested):
Personally, I would skip this altogether.

schildbach commented:
I converted this to draf for now and will try to rework this a bit. I rebased on current master.

As a first change, the read and write helpers are now using ByteBuffer directly. Since we're dealing with signed long, it's no big deal.

schildbach commented:
I rebased this, and replaced the `asBuffer()` helper with `serialize()` like with the other data classes.

msgilligan commented on core/src/main/java/org/bitcoinj/base/Coin.java:
```diff
@@ -354,6 +372,27 @@ public BigDecimal toBtc() {
+        ByteBuffer buf = ByteBuffer.allocate(BYTES);
+        return write(buf).array();
+    }
+
```
I still think that `serialize()` should not be using an intermediate `ByteBuffer` here, but I suppose you could call that "premature optimization", so I'm going to go ahead an approve the PR anyway.

schildbach replied:
I don't see a way to prevent this, if we want to use `ByteBuffer.putLong(value)` and *not* do our own bitshuffling.

msgilligan reviewed (approved):
Approved, with one suggestion that we can consider later.

schildbach commented on core/src/main/java/org/bitcoinj/base/Coin.java:
```diff
@@ -354,6 +369,18 @@ public BigDecimal toBtc() {
+        ByteUtils.writeInt64LE(this.value, buf);
+        return buf;
+    }
+
```
We'll remove the unused helpers before release.

schildbach commented:
Merged.
